test/integration/test-amp-carousel.js

```diff
@@ -14,249 +14,351 @@
  * limitations under the License.
  */
 
-import {AmpEvents} from '../../src/amp-events';
-import {
-  createFixtureIframe,
-  expectBodyToBecomeVisible,
-} from '../../testing/iframe';
+const config = describe.configure().ifNewChrome();
+config.run('amp-carousel', function() {
+  this.timeout(10000);
 
-describe.skip('integration amp-carousel', () => {
+  const extensions = ['amp-carousel'];
 
-  let fixture;
-  beforeEach(() => {
-    return createFixtureIframe('test/fixtures/carousels.html', 1000)
-        .then(f => {
-          fixture = f;
-        });
-  });
+  const carouselSingleImage = `
+  <amp-carousel width=400 height=300 id="carousel-1">
+    <amp-img src="https://lh3.googleusercontent.com/pSECrJ82R7-AqeBCOEPGPM9iG9OEIQ_QXcbubWIOdkY=w400-h300-no" layout=fill></amp-img>
+  </amp-carousel>
+  `;
 
-  it('should show the body in carousel test', () => {
-    return expectBodyToBecomeVisible(
-        fixture.win, window.ampTestRuntimeConfig.mochaTimeout);
-  });
+  describes.integration('type=carousel with single image', {
+    body: carouselSingleImage,
+    extensions,
+  }, env => {
+    let document;
+
+    beforeEach(() => {
+      document = env.win.document;
+    });
+
+    it('should be present', () => {
+      expect(document.querySelectorAll('amp-carousel')).to.have.length.above(0);
+    });
+
+    it('should not have the buttons visible ' +
+        'when amp-mode-mouse class is not on body', () => {
+      document.body.classList.remove('amp-mode-mouse');
+      return Promise.resolve().then(timeout(2000))
```

Comment:
Can we do better than "wait for 2 seconds" here? What is it actually waiting for?